### PR TITLE
Editor: Font sizes now allow decimals and updated text presets font sizes. 

### LINKED ISSUE #4419: Text size doesn't accept fractional sizes
## Bug Description

Testing #3639 noticed that our editor doesn't accept fractional text sizes in the text size input box. We should fix this to allow fractional text input. Once that's done, we should also revisit the spec for text presets and ensure fractional sizes are properly set (e.g., H3 should be 21.6, Caption should be 13.5)

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by mariano-formidable**
## Summary

This is a small PR that enables float values for font-size in the Editor.  It also updates the text presets to make sure the font-size of each preset matches the figma docs requirements.

## User-facing changes

- The font-size for each text preset matches the [figma docs](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=710%3A0).
- Text blocks now allow for decimal points in font-size:

NOTE: I added a prop for setting `minWidth` on the dropdown so I could shrink the font-weight dropdown a little bit so fonts with decimal points won't get clipped.

## Testing Instructions

Decimal font-size:
1.) Open up a story in the Editor.
2.) Click on existing text block or add a new text block.
3.) Change the font size to 12.5 or some other decimal'd number and it should apply like normal.

Text Presets:
Add each preset and check that:
"Heading 1" has font-size of `36`
"Heading 2" has font-size of `27`
"Heading 3" has font-size of `21.6`
"Paragraph" has font-size of `18`
"Caption" has font-size of `13.5`
"Overline" has font-size of `12`

Font-size chart from figma design doc:

Fixes #4419

### CODE REVIEW

**Review by BrittanyIRL — APPROVED** [review]
Works well for me, decimals are showing up. text presets are accurately sized.

**Review by maxyinger — APPROVED** [review]
lgtm

File: assets/src/edit-story/components/panels/textStyle/font.js [hunk 1]
```diff
@@ -168,6 +168,7 @@ function FontControls({ selectedElements, pushUpdate }) {
         {fontWeights && (
           <>
             <DropDown
+              minWidth={140}
```
Comment (miina) [inline assets/src/edit-story/components/panels/textStyle/font.js]: Hmm, if the flex-basis of the font-size stays the same, would reducing the size of the font-weight input actually help with giving more room to the font-size input?

Just tested locally as well and in my case at least, the only thing I see is just more room between the font-weight and the font-size inputs. Should the font-weight perhaps have `flex: 1` instead then to just take the remaining room? This, however, would mean that the list width would differ from the input width...

How it looks locally -- the same width for the font-size, just extra room between:
Screenshot 2020-10-02 at 10 12 26

vs on `main`:
Screenshot 2020-10-02 at 10 15 50

File: assets/src/edit-story/components/library/panes/text/karma/textPane.karma.js [hunk 2]
```diff
@@ -68,7 +68,7 @@ describe('CUJ: Creator can Add and Write Text: Consecutive text presets', () =>
       await waitFor(() => {
         expect(stripHTML(element.content)).toEqual(content);
         const preset = PRESETS.find(({ title }) => name === title);
-        expect(element.y).toEqual(preset.element.y);
```
Comment (miina) [inline assets/src/edit-story/components/library/panes/text/karma/textPane.karma.js:71]: Maybe add a test for using a float to the font-size field as well (e.g. a unit test in addition to `'should select font size'`)?

**Comment by carloskelly13** [comment]
Agree with Miina on the tests and the gap between the style and size in the panel.

File: assets/src/edit-story/components/form/dropDown/index.js [hunk 3]
```diff
@@ -36,13 +36,13 @@ import {
...
 import Popup, { Placement } from '../../popup';
-import DropDownList from './list';
+import DropDownList, { DEFAULT_MIN_WIDTH } from './list';
 
 /* same min-width as ListContainer */
 const DropDownContainer = styled.div`
   display: flex;
   flex-direction: column;
-  min-width: 160px;
+  min-width: ${({ minWidth }) => minWidth || DEFAULT_MIN_WIDTH}px;
```
Comment (carloskelly13) [inline assets/src/edit-story/components/form/dropDown/index.js]: I do wonder if `flex: 1` here is the better option. Visually that seems to make it fill the container correctly without having to resort to numeric metrics. I'd imagine that most of the dropdowns are inside containers so I can't see this having ripple effects. Alternatively we could have a prop that `flex: 1` like `flexContainer` or something.

File: assets/src/edit-story/components/panels/textStyle/font.js (hunk 1 above)
Comment (mariano-formidable) [reply assets/src/edit-story/components/panels/textStyle/font.js]: Good call, I'll switch to flex width's so the dropdown can fill in the space as needed.  When I was testing, I didn't have the editor at full screen so I didn't notice the gap, for me it looked like this:


But as soon as I went full screen and the panel width increased, then boom, the gap widened 😱 I'll fix.

File: assets/src/edit-story/components/form/dropDown/index.js [hunk 4]
```diff
@@ -34,18 +34,18 @@ import {
...
 import Popup, { Placement } from '../../popup';
-import DropDownList from './list';
+import DropDownList, { DEFAULT_MIN_WIDTH } from './list';
 
 /* same min-width as ListContainer */
 const DropDownContainer = styled.div`
   display: flex;
   flex-direction: column;
-  min-width: 160px;
+  flex: 1;
```
Comment (mariano-formidable) [inline assets/src/edit-story/components/form/dropDown/index.js:45]: Setting `flex: 1` seemed to fix the issue, now the dropdown container expands with the space:

File: assets/src/edit-story/components/form/dropDown/index.js (hunk 3 above)
Comment (mariano-formidable) [reply assets/src/edit-story/components/form/dropDown/index.js]: Seems like it works! 🎉

**Comment by mariano-formidable** [comment]
@carlos-kelly @miina I believe I addressed your comments, this PR is ready for re-review 😎 (with added tests).

**Review by miina — APPROVED** [review]
LGTM

**Review by BrittanyIRL — APPROVED** [review]
👍 💯  LGTM, works as expected. Verified no extra gap depending on viewport size

File: assets/src/edit-story/units/dimensions.js [hunk 5]
```diff
@@ -44,7 +44,7 @@ export function dataPixels(v) {
  * @return {number} The font size for the specified "em" value.
  */
 export function dataFontEm(v) {
-  return dataPixels(v * DEFAULT_EM);
+  return Number((v * DEFAULT_EM).toFixed(1));
```
Comment (barklund) [inline assets/src/edit-story/units/dimensions.js:47]: Doesn't this change existing font sizes?

**Review by barklund — APPROVED** [review]
This seems to work great.

**Comment by csossi** [comment]
Verified in QA

**Pull request merged**